src/tfr_control/src/test_client.cpp

```diff
@@ -0,0 +1,97 @@
+ *          ActionServer clients, but does not include every function available
+ *          to a client. If you are adapting this code for your system, please
+ *          read the ROS API documentation on actionlib::SimpleActionClient<> at 
+ *          http://docs.ros.org/api/actionlib/html/classactionlib_1_1SimpleActionClient.html
+ * 
+ *          This file includes <tfr_msgs/ExampleAction.h>, which is one of seven headers
+ *          built by catkin from `tfr_msgs/action/Example.action`:
+ *              devel/include/tfr_msgs/ExampleAction.h
+ *              devel/include/tfr_msgs/ExampleActionFeedback.h
+ *              devel/include/tfr_msgs/ExampleActionGoal.h
+ *              devel/include/tfr_msgs/ExampleActionResult.h
+ *              devel/include/tfr_msgs/ExampleFeedback.h
+ *              devel/include/tfr_msgs/ExampleGoal.h
+ *              devel/include/tfr_msgs/ExampleResult.h
+ * 
+ ***************************************************************************************/
+
+#include <tfr_msgs/ArmMoveAction.h> // Note: "Action" is appended
+#include <actionlib/client/simple_action_client.h>
+#include <std_msgs/Float64.h>
+#include <vector>
+#include <map>
+
+typedef actionlib::SimpleActionClient<tfr_msgs::ArmMoveAction> Client;
+
+// Called once when the goal completes
+void finished(const actionlib::SimpleClientGoalState& state, const tfr_msgs::ArmMoveResultConstPtr& result)
+{
+    // If this goal didn't succeed (there's multiple fail states in the enum so
+    // it's easier to check for this)
+    if (state.state_ != actionlib::SimpleClientGoalState::StateEnum::SUCCEEDED) {
+        ROS_WARN("Error planning/executing motion.");
+        return;
+    }
+
+    ROS_INFO("Planning and execution succeeded.");
+}
+
+int main(int argc, char** argv)
+{
+    // The list of positions to travel to, in order
+    std::vector<std::string> position_names = { "ready", "dig", "scoop", "out", "dump", "release", "invalid" };
+
+    // Basic ROS setup
+    ros::init(argc, argv, "example_client");
+    ros::NodeHandle n;
+
+    // Needed to spin while we're waiting for a blocking call later (waiting for
+    // server finished response)
+    ros::AsyncSpinner spin(1);
+    spin.start();
+    
+    // Set up the action server and connect
+    Client client("move_arm", true);
+
+    ROS_INFO("Connecting to action server...");
+    client.waitForServer();
+    ROS_INFO("Connection established with server.");
+
+    // Loop through all of the positions to travel to and then exit
+    for (auto iter = position_names.begin(); iter != position_names.end(); iter++) {
+        // Get the name of the parameter in the positions namespace
+        std::string param_name = "positions/" + *iter;
+        ROS_INFO("Loading value: %s", param_name.c_str());
+
+        // Load the parameters (an array of doubles which represent joint angles)
+        std::vector<double> angles;
+        if (!n.getParam(param_name, angles)) {
```

Comment:
In the future if these are parameters for just that node, we should prefer private namespace scope, to avoid namespace pollution.

Reply:
http://wiki.ros.org/roscpp/Overview/Parameter%20Server
Read about how to access and use private parameters.

Reply:
However again, test code so not a required change

Reply:
Good note, I'll do that for the future. Might do it now just for a test/practice.

Reply:
Implemented this in the upcoming commit.